canonical-web-and-design/react-components
# Add generic type for spreading props

## Done

- Added `PropsWithSpread` type that ensures that defined component props are never overwritten by a subsequent prop spread
- Added descriptions to some base types

## QA
### QA steps

- Pull this branch and open `SearchBox.tsx` in VSCode
- Hover over the props in the component function argument and check that the `onChange`, `onSubmit` and `value` props are typed according to the `Props` definition.
